pkg/volume/empty_dir/quota.go

```diff
@@ -0,0 +1,128 @@
+package empty_dir
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// An object for applying quota to an empty dir volume.
+type quotaApplicator interface {
+	// Apply the quota to the given EmptyDir path:
+	Apply(dir string, medium api.StorageMedium, pod *api.Pod) error
+}
+
+type xfsQuotaApplicator struct {
+}
+
+func (xqa *xfsQuotaApplicator) Apply(dir string, medium api.StorageMedium, pod *api.Pod, fsGroup *int64, quota int) error {
+	if medium == api.StorageMediumMemory {
+		glog.V(3).Infof("skipping quota application due to memory storage medium.")
+		return nil
+	}
+	if !xqa.isXfs(dir) {
+		glog.V(3).Infof("%s is not on an XFS filesystem, skipping quota application.")
```

Comment:
Quota right now is going to be admin policy applied to the node.  If
the policy is enabled, failure to apply the policy should definitely
be an error.  When it moves to the pod, if the node can't satisfy the
policy, that should also be an error.